src/service/cid-impl.js

```diff
@@ -382,30 +400,100 @@ function setCidCookie(win, scope, cookie) {
   });
 }
 
+/**
+ * Sets a new CID backup in Storage
+ * @param {!./ampdoc-impl.AmpDoc} ampdoc
+ * @param {string} cookieName
+ * @param {string} cookie
+ */
+function setCidBackup(ampdoc, cookieName, cookie) {
+  Services.storageForDoc(ampdoc).then((storage) => {
```

Comment:
I know we only call this function within non proxy origin. But I think it's safer to check whether storage API is used here for extra safety.

Reply:
Good call!